api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -159,17 +159,32 @@ type VolumeReplicationGroupSpec struct {
 	KubeObjectProtection *KubeObjectProtectionSpec `json:"kubeObjectProtection,omitempty"`
 }
 
+const ReservedBackupName = "use-backup-not-restore"
+
+type RecipeSpec struct {
+	//+optional
+	Name *string `json:"name,omitempty"`
+
+	//+optional
+	Workflow *WorkflowSpec `json:"workflow,omitempty"`
+}
+
+type WorkflowSpec struct {
+	//+optional
+	CaptureName *string `json:"captureName,omitempty"`
+
+	//+optional
+	RecoverName *string `json:"recoverName,omitempty"`
+}
+
 type KubeObjectProtectionSpec struct {
 	// Preferred time between captures
 	//+optional
 	//+kubebuilder:validation:Format=duration
 	CaptureInterval *metav1.Duration `json:"captureInterval,omitempty"`
 
 	//+optional
-	CaptureOrder []KubeObjectsCaptureSpec `json:"captureOrder,omitempty"`
-
-	//+optional
-	RecoverOrder []KubeObjectsRecoverSpec `json:"recoverOrder,omitempty"`
+	Recipe *RecipeSpec `json:"recipe,omitempty"`
```

Comment:
I know the change to support separate capture and recover workflows is deferred from this PR, however I think this field name should be changed to `RecipeRef` to differentiate it from the recipe and align with the API [convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field)
> The name of the reference field should be of the format "{field}Ref", with "Ref" always included in the suffix.

I also suggest adding `Ref` to the type name, e.g., `RecipeRefSpec` or simply `RecipeRef`.

Reply:
Based on this and [Shyam's comment](https://github.com/RamenDR/ramen/pull/675#discussion_r1108904187), I'm unsure which of these versions is what is being proposed:

Option 1:
```yaml
spec:
  kubeObjectProtection:
    recipeRef:
      name:
      captureWorkflowName:
      restoreWorkflowName:
      volumeGroupName:
```

Option 2:
```yaml
spec:
  kubeObjectProtection:
    recipe:
      recipeRef:
      captureWorkflowName:
      restoreWorkflowName:
      volumeGroupName:
```

Or something else?

Reply:
Option 1!